content/advanced/fine-grained-reactivity.mdx

```diff
@@ -0,0 +1,222 @@
+# Understanding Fine-Grained Reactivity
+
+Reactivity is a programming pattern that lets you set up behavior that "reacts" to data changes automatically.
+
+For example, you might make a shopping cart with a label that displays the total cost of the items. With reactivity, you can tell that label to automatically update whenever an item gets added or removed.
+
+This way, you only have to manage the data. Once you wire up the UI elements to the data, the UI updates are handled for you.
+
+To understand how Solid handles fine-grained reactivity, a basic understanding of reactive signals and effects is required. A deeper understanding of reactivity can help you optimize your code, debug issues, and build your own extensions to the reactive system.
+
+## Reactive Primitives
+
+The most important primitives in Solids reactivity system are signals and effects. The sections below will explain what signals and effects are, and show a simple of implementation of them.
+
+<Aside type="info" title="Other reactive pimitives">
+    Solid has other reactive primitive, but they are derived from signals and effects. For example:
+
+* Stores are trees of signals
+* Memos are signals that update like effects
+* Resources are signals that update when data is fetched from a server
+* Render effects are effects that initially run earlier
```

Comment:
Maybe say that render effects are a variant of effect used primarily for rendering.

Reply:
> Render effects are a variant of effects with an earlier initial execution, used to implement the rendering process